lib/simplecov.rb

```diff
@@ -54,6 +54,40 @@ def start(profile = nil, &block)
       start_coverage_measurment
     end
 
+    #
+    # Collate a series of SimpleCov result files into a single SimpleCov output.
+    # You can optionally specify configuration with a block:
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"]
+    #    OR
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"], 'rails' # using rails profile
+    #    OR
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"] do
+    #     add_filter 'test'
+    #   end
+    #    OR
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"], 'rails' do
+    #     add_filter 'test'
+    #   end
+    #
+    # Please check out the RDoc for SimpleCov::Configuration to find about available config options
+    #
+    def collate(result_filenames, profile = nil, &block)
+      load_profile(profile) if profile
+      configure(&block) if block_given?
+      self.running = true
+      self.pid = Process.pid
+
+      results = result_filenames.map { |filename| JSON.parse(File.read(filename)) || {} }.reduce(&:merge).map do |command_name, data|
+        # Re-create each included instance of SimpleCov::Result from the stored run data.
+        SimpleCov::Result.from_hash(command_name => data)
+      end
+
+      # Use the ResultMerger to produce a single, merged result, ready to use.
+      @result = SimpleCov::ResultMerger.merge_results(*results)
+      SimpleCov::ResultMerger.store_result(@result) if @result
+      @result
```

Comment:
I needed to add the above two lines to get tests passing because I noticed that the merged coverage resultset was not being written to disk. That surprised me though because I thought some people were using this, but it seems to me that without these lines this feature can't work (as the cucumber scenario was showing). Maybe I'm missing something? :thinking:

Reply:
Right now your guess is as good as mine I haven't read that code in a long while and need to fresh up on it again :sweat_smile:

Reply:
I _think_ the reason is that it's written to disk in the `at_exit` hook, which the tests doesn't really run. But I need to dig in deeper.

Reply:
I think this was an oversight on my part - we're relying on plugins' output, which are written regardless of the merged result set JSON being stored. Making it write this makes sense to me. Good catch!